src/v/cloud_storage_clients/client_pool.cc

```diff
@@ -442,7 +442,7 @@ void client_pool::populate_client_pool() {
     _cvar.signal();
 }
 
-client_pool::http_client_ptr client_pool::make_client() const {
+client_pool::http_client_ptr client_pool::make_client() const noexcept {
```

Comment:
That’s my assumption. The code using this method doesn’t seem to be exception safe so I made make_client fail hard rather than cause surprising behaviour.

Reply:
One example is when we’re borrowing from another shard. If make_client throws an exception there, then we would never return the borrow.

Reply:
got it, so it's the code in a different place is not exception safe if this method could throw